packages/core/src/config.ts

```diff
@@ -531,6 +542,85 @@ const composeFormatConfig = (format: Format): RsbuildConfig => {
   }
 };
 
+const resolveShims = (shims?: Shims) => {
+  const resolvedShims = {
+    cjs: {
+      'import.meta.url': true,
+    },
+    esm: {
+      __filename: true,
+      __dirname: true,
+      require: false,
+    },
+  };
+
+  if (!shims) {
+    return resolvedShims;
+  }
+
+  if (shims.cjs) {
+    if (typeof shims.cjs === 'boolean') {
+      if (shims.cjs === true) {
+        resolvedShims.cjs['import.meta.url'] = true;
+      } else {
+        resolvedShims.cjs['import.meta.url'] = false;
+      }
```

Comment:
only need to deal when `shims.cjs === false`? `resolvedShims.cjs['import.meta.url']` default to true